app/src/main/java/org/gnucash/android/ui/report/ReportsOverviewFragment.java

```diff
@@ -178,14 +175,13 @@ protected void generateReport() {
      * @return {@code PieData} instance
      */
     private PieData getData() {
-        String mCurrencyCode = GnuCashApplication.getDefaultCurrencyCode();
         PieDataSet dataSet = new PieDataSet(null, "");
         List<String> labels = new ArrayList<>();
         List<Integer> colors = new ArrayList<>();
         for (Account account : mAccountsDbAdapter.getSimpleAccountList()) {
             if (account.getAccountType() == AccountType.EXPENSE
                     && !account.isPlaceholderAccount()
-                    && account.getCurrency() == Currency.getInstance(mCurrencyCode)) {
+                    && account.getCommodity() == mCommodity) {
```

Comment:
Java Currency class uses Singletons, so the `==` operator will work for them. But I don't think Commodity works this way at the moment. It does not use singletons
That means the `equals()` method should probably be used here instead. It has been properly implemented for the Commodity class.